draft-ietf-quic-transport.md

```diff
@@ -322,175 +322,356 @@ all field sizes are in bits.  When discussing individual bits of fields, the
 
-Version negotiation ({{version-negotiation}}) is performed using packets that
-have the VERSION bit set.  This bit is always set on packets that are sent prior
-to connection establishment.  When receiving a packet that is not associated
-with an existing connection, packets without a VERSION bit MUST be discarded.
+The packet layout is the same for all long-header packet types, but the
+semantics of the fields are specific to each packet type.  Type-specific
+semantics are described in {{version-negotiation-packet}},
+{{public-reset-packet}}, {{cleartext-packet}}, and {{encrypted-packet}}.
 
-Implementations MUST assume that an unsupported version uses an unknown packet
-format.
 
-Between different versions the following things are guaranteed to remain
-constant are:
+## Short Header
 
-* the location and size of the Flags field,
+```
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+
+|0|C|K|  Type   |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                                                               |
++                   Connection ID (optional)                    +
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                     Packet Number (1/2/4)                     |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                       Encrypted Payload                     ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+```
 
-* the location and value of the VERSION bit in the Flags field,
+The short header can be used after the version and 1-RTT keys are negotiated.
+This header form has the following fields:
 
-* the location and size of the Connection ID field, and
+Header Form:
 
-* the Version (or Supported Versions, {{version-negotiation-packet}}) field.
+: The most significant bit (0x80) of the first octet (octet 0) of a packet is
+  the header form.  This bit is set to 0 for the short header.
 
-All other values MUST be ignored when processing a packet that contains an
-unsupported version.
+Connection ID Flag:
 
+: The second bit (0x40) of octet 0 indicates whether the connection ID field is
+  present.  If set to 1, then the connection ID field is present; if set to 0,
+  the connection ID field is zero length.
 
-## Regular Packets
+Key Phase Bit:
 
-Each Regular packet contains additional header fields followed by an encrypted
-payload, as shown below:
+: The third bit (0x20) of octet 0 indicates the key phase, which allows a
+  recipient of a packet to identify the packet protection keys that are used to
+  protect the packet.  See {{QUIC-TLS}} for details.
+
+Short Packet Type:
+
+: The remaining 5 bits of octet 0 include one of 32 packet types.
+  {{short-packet-types}} lists the types that are defined for short packets.
```

Comment:
Yeah, there will be holes or overlap between the header types. I don't see the value in unifying since there's got to be a branch before reading the types about which bits to read from the packet anyways, and two small maps aren't really that expensive to maintain.